python/tank/platform/constants.py

```diff
@@ -175,6 +175,12 @@
 # the name of the shell engine
 SHELL_ENGINE = "tk-shell"
 
+# valid characters for a template key name
+TEMPLATE_KEY_NAME_REGEX = "[a-zA-Z_ 0-9\.]+"
```

Comment:
It is shared by two files, that's why I moved it out. I agree that it shouldn't be in `tank.platform`, that `tank` should have its own constants file probably - but that seemed like outside the scope of this change. I am not sure I understand the point about the cleanup - can't you just grep the code?